app/controllers/search_controller.rb

```diff
@@ -1,26 +1,33 @@
 class SearchController < ApplicationController
 
-  before_action :ensure_user_area
+  before_action :ensure_user_municipality
 
   def show
-    @demands = user_area.demands.to_a
+    @demands = user_municipality.demands.to_a
 
     if params[:q].present?
       # TODO: replace this with a proper search engine. elasticsearch?
       query = params[:q].split(' ').join('|')
 
-      demands = user_area.demands.advanced_search(solution: query)
+      demands = user_municipality.demands.advanced_search(solution: query)
 
-      demands_by_problem = user_area
+      demands_by_problem = user_municipality
                    .problems
                    .includes(:demands)
-                   .advanced_search(name: query).map { |problem| problem.demands.find { |demand| demand.area == user_area }}.compact
+                   .advanced_search(name: query).map { |problem| problem.demands.find { |demand| demand.area == user_municipality }}.compact
 
       @demands = Set.new(demands)
                    .merge(demands_by_problem).to_a
 
     end
 
+    if params[:scope] == 'county'
```

Comment:
Should this be in an `elsif` of line 8? (`params[:q].present?`)

Reply:
Sure. The whole search functionality needs to be refactored. I hooked up client-side searching against the current list displayed, but we probably want full fledged search and will have to combine params rather than `elsif` them. I have it as an issue to do.

Reply:
I probably wouldn't bother with anything client-side for this app, for now. It's a server-rendered turbolinks app, doing that goes against the grain of the current tech stack :-D